pylint/checkers/typecheck.py

```diff
@@ -1720,9 +1720,18 @@ def visit_subscript(self, node):
             return
 
         inferred = safe_infer(node.value)
+
         if inferred is None or inferred is astroid.Uninferable:
             return
 
+        if inferred.decorators:
+            first_decorator = helpers.safe_infer(inferred.decorators.nodes[0])
```

Comment:
IMHO we should not restrict to the first decorator but instead emit the message if none of the decorators enable the support of the protocol.

Reply:
The type of the object will be of the type of the outer decorator (the "first" one from top to bottom), as the result of other decorators being just passed as an argument to the last, so I though it was enough.

We have nothing prooving a decorator will return the object that it receive (most will not, most will just return a new callable).

So I think we should really just look at the first one (hope `[0]` is the top one, the outer one, I don't know how to call it) hope I did not get it wrong?

(When I say "outer" I think of decorators as applied manually:

```
@a
@b
@c
def foo(): ...
```

being `foo a(b(c(foo)))`, so when I say "outer decorator" I mean `a` in this case, I don't know if there's a better wording?)

Reply:
@JulienPalard i think you are right. Let's do as you think.